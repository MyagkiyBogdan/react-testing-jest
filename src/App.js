function App() {
  return (
    <div className="App">
      <h1>Hello World</h1>
      <button>Click me</button>
      <input type="text" placeholder="name" />
    </div>
  );
}

export default App;
