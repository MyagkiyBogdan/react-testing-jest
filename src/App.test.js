import { render, screen } from '@testing-library/react';
import App from './App';

describe('App', () => {
  test('render App test', () => {
    render(<App />);
    const helloWorldElem = screen.getByText(/hello world/i); // регулярное выражения для поиска текста с игнорированием регистра
    const btn = screen.getByRole('button');
    const input = screen.getByPlaceholderText(/name/i);
    expect(helloWorldElem).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    // screen.debug();
    expect(input).toMatchSnapshot();
  });
});
