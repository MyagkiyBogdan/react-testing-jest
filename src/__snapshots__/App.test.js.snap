// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render App test 1`] = `
<input
  placeholder="name"
  type="text"
/>
`;

exports[`render App test 1`] = `
<input
  placeholder="name"
  type="text"
/>
`;
